x/ecocredit/basket/msg_create_test.go

```diff
@@ -26,45 +26,103 @@ func TestMsgCreateValidateBasic(t *testing.T) {
+		{"curator-1",
+			MsgCreate{Curator: "wrong"},
+			"malformed curator address"},
+		{"name-1",
+			MsgCreate{Curator: a, Name: ""}, "name must not be empty"},
+		{"name-2",
+			MsgCreate{Curator: a, Name: randstr.String(nameMaxLen + 1)},
+			"name must not be empty and must not be longer than"},
+		{"name-3",
+			MsgCreate{Curator: a, Name: name, DisplayName: ""},
+			"display_name must be between"},
+		{"name-4",
+			MsgCreate{Curator: a, Name: name, DisplayName: randstr.String(displayNameMaxLen + 1)},
+			"display_name must be between"},
+		{"exponent-1",
+			MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax + 1},
+			"exponent must not be bigger than"},
+		{"credity_type-1",
+			MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax},
 			"credit_type_name must be defined"},
-		{MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: randstr.String(creditTNameMaxLen + 1)},
+		{"credity_type-2",
+			MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: randstr.String(creditNameMaxLen + 1)},
 			"credit_type_name must not be longer"},
-		{MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, MinStartDate: start},
+		{"date_criteria-1",
+			MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: &DateCriteria{}},
+			"unsupported date_criteria value"},
+		{"allowed_classes-1",
+			MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: start},
 			"allowed_classes is required"},
-		{MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, MinStartDate: start, AllowedClasses: []string{"class1", ""}},
+		{"allowed_classes-2", MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: start, AllowedClasses: []string{"class1", ""}},
 			"allowed_classes[1] must be defined"},
-		{MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, MinStartDate: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{Denom: "1a"}}},
+		{"fee-1", MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{Denom: "1a"}}},
 			"invalid denom"},
-		{MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, MinStartDate: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{"aa", sdk.NewInt(-1)}}},
+		{"fee-2", MsgCreate{Curator: a, Name: name, DisplayName: dName, Exponent: exponentMax, CreditTypeName: creditName, DateCriteria: start, AllowedClasses: classes, Fee: sdk.Coins{sdk.Coin{"aa", sdk.NewInt(-1)}}},
```

Comment:
It would be nice to use consistent formatting and to make these cases more readable.

Reply:
yes, will move them to a new line